pyproject.toml

```diff
@@ -414,6 +414,7 @@ ignore = [
   "SIM118", # remove .keys() calls from dictionaries
   "SIM300", # yoda conditions
   "UP037", # remove quotes from type annotation
+  "UP007", # Optional[str] -> str | None
```

Comment:
Our typehint evaluation doesn't support the new syntax for `python<3.10`. To support that we would need to rewrite the AST before evaluating, but I don't think that worth the effort. 

Nonetheless if we would like to support it we can use the [solution from typeguard](https://github.com/agronholm/typeguard/blob/master/src/typeguard/_union_transformer.py).